### DISCUSSION TITLE
Unexpected renaming of slide XML files after SaveAs

### DISCUSSION THREAD

**Discussion by shanaki** [ShapeCrawler/ShapeCrawler#266] [Q&A]
When open presentation slide by IPresentation.Open() and then just save it by IPresentation.SaveAs().
Rename suffix of the saved file to *.zip, then unzip and see ppt/slides folder.
slide*.xml filename are changed like slide25555.xml, slide67777.xml. Origin ppt contains slide1.xml, slide2.xml and so on.
Why slide xml filename are changed by Save()? Is this intentional behavior? Or, any way to prevent it?
![sc](https://user-images.githubusercontent.com/30141376/186626604-5b8b6eb7-4c65-44e8-bf8e-f8b33390f3fb.png)

**Comment by AdamShakhabov** [comment]
Could you share more details to reproduce this file name corruption? What data have you changed before `.Save()`?

For example, I wasn't able to reproduce this when I just open/save a presentation:

```C#
using var pres = SCPresentation.Open("test.pptx", true);
pres.Save();
```

**Reply by shanaki** [reply]
Thanks for reply. I mistook to wrote the step. To reproduce the issue, please use SaveAs().

**Reply by AdamShakhabov** [reply]
I can reproduce this. Yeah, looks bad.

@shanaki, thank you for reporting this.

**Reply by shanaki** [reply]
So, is this a kind of a bug to be submitted an issue?

**Reply by AdamShakhabov** [reply]
I think it is not a _functional_ bug, but it must be fixed.

I've created issue item for this: https://github.com/ShapeCrawler/ShapeCrawler/issues/271

**Reply by shanaki** [reply]
Thanks for creating issue.
When I open the "save as" presentation file in PowerPoint and then just save it, rename suffix to zip, then unzip it. go to pptx/slide. slide*.xml filename is slide1.xml, slide2.xml.... reverted to expected filenames. This is interesting.
